### PR TITLE
Multi arch docker volume plugin

### PR DESCRIPTION
**Pull request by tstachl**
#### What is the purpose of this change?

Docker volume plugins, like Docker containers, differ based on the architecture of the host. The purpose of this change is to publish docker volume plugins for all supported platforms: amd64, arm64, 386, and armv7

If you're trying to install the plugin on a arm64 device, it throws an error like this:

> Note: I tested it with @ivandeex's repository since, I'm guessing, the rclone user doesn't have the plugin pushed yet.

```bash
$ docker plugin install ivandeex/docker-volume-rclone:latest args="-v" --alias rclone --grant-all-permissions
latest: Pulling from ivandeex/docker-volume-rclone
Digest: sha256:360c31d38e6360b0bcb4aea06b2281a3406ba07330304dec98fdf7ec18893d30
54daab5d3884: Complete
Error response from daemon: dial unix /run/docker/plugins/5e0faf434bf7f10ee3b1169c00604bee06e9fb22b67f38cc024019545ad982f9/rclone.sock: connect: no such file or directory
```

Also, I tested this pull request with arm64 and amd64. It would be great if someone could test the other two platforms as well.

#### Was the change discussed in an issue or in the forum before?

No, the issue has not been discussed yet.

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by ivandeex** [comment]
With my current load I cannot do much before this October

**Comment by ivandeex** [comment]
Hi @tstachl
Before I can review your changes I squashed the commits on your branch and rebased on the latest rclone master.
Unfortunately you submitted a branch belonging to 3rd-party user @pilina rather than @tstahl :(
This locks me out of branch access.
Could you please do one of the following?
- grant me with write access on this branch
- accept the PR https://github.com/pilina/rclone/pull/2
- hard-reset this branch to my branch https://github.com/ivandeex/rclone/tree/multiarch-docker-plugin
- resubmit pull request from a branch owned by @tshahl
- let me submit another pull request based on this one
Thanks

**Review by ivandeex — CHANGES_REQUESTED** [review]
Please see my comments inline.
Also I'd like to make a few notes not belonging in the diff:

- we need to update documentation to let users know the new plugin image tags and how to install...
- before this change we had `latest` and version tags, now we have architecture/version tags.
   it's a breaking change and must go to changelog
- all went well when I tried new Makefile with Docker Desktop on Windows,
  but on Linux I had to install QEMU support as explained [here](https://github.com/docker/buildx#building-multi-platform-images) or [here](https://medium.com/@artur.klauser/building-multi-architecture-docker-images-with-buildx-27d80f7e2408).
   This check should be automatic, let's add it to Makefile
   (if the code grows too much lets factor it out of Makefile to a script in `bin/`)...

In a moment I myself will submit another commit(s) addressing the review comments.
Please wait... 😪

File: Makefile [hunk 1]
```diff
@@ -264,28 +264,49 @@ PLUGIN_IMAGE_NAME ?= docker-volume-rclone
...
+	rm -rf ${PLUGIN_BUILD_DIR}
 
-docker-plugin-create:
-	docker plugin rm -f ${PLUGIN_IMAGE} 2>/dev/null || true
-	docker plugin create ${PLUGIN_IMAGE} ${PLUGIN_BUILD_DIR}
+	docker buildx create --use \
+		--platform ${PLUGIN_PLATFORM} \
+		--name ${PLUGIN_CONTAINER}
+
+	docker buildx build --no-cache --pull \
```
[the remark this answers is not in the document]
Comment (ivandeex) [reply Makefile:284]: Another good side of `buildx` is its ability to build multiple outputs (or parts of multi-arch image) in parallel greatly reducing build time. We would not gain a lot for docker plugin from concurrent build because now it simply adds a single binary to the `apline` image and adjusts a few settings. However the rclone release build would gain a lot. That step is based on an external buildx-based package [ilteoood/docker_buildx](https://github.com/ilteoood/docker_buildx/releases) so I think we should monitor for new versions and upgrade when it starts to support concurrency.

File: cmd/serve/docker/contrib/plugin/Dockerfile [hunk 2]
```diff
@@ -2,22 +2,11 @@ ARG BASE_IMAGE=rclone/rclone:latest
...
-    CGO_ENABLED=0 \
-    GOARCH=$(echo ${TARGET_PLATFORM} | cut -d '/' -f2) \
-    make rclone
-
 # plugin image
-FROM ${BASE_IMAGE}
-
-COPY --from=BUILD_ENV /src/rclone /usr/local/bin/rclone
+FROM ${BASE_IMAGE} as binaries
+FROM alpine:latest
```
[the remark this answers is not in the document]
Comment (ivandeex) [reply cmd/serve/docker/contrib/plugin/Dockerfile:7]: However you forgot to install `FUSE` in the image. Plugin will refuse to mount 😢

**Comment by ivandeex** [comment]
@tstachl
I addressed all review comments in a new commit on the branch https://github.com/ivandeex/rclone/commits/multiarch-docker-plugin, 
then temporarily [pointed build job at my docker hub](https://github.com/ivandeex/rclone/commit/db08c61ec6a9f834a9720f69820ba91a918150b1) on the [CI branch](https://github.com/ivandeex/rclone/commits/ci-multiarch-docker-plugin) and [successfully performed](https://github.com/ivandeex/rclone/actions/workflows/build_publish_release_docker_image.yml) a few test builds. The results are [available on my docker hub](https://hub.docker.com/repository/docker/ivandeex/docker-volume-rclone/tags?page=1&ordering=last_updated) and can be installed with `docker plugin install ivandeex/docker-volume-rclone:arm64`.
Please play with it, review my changes and let's think about merging (can you solve the permission issue?)

@ncw
This PR is ready for final review. Please take a look in the code.
I specifically want your opinion on two topics:
- I removed plugin build for `386` as I think it's exotic from the docker point of view. We build for `amd64`, `arm64`, and `arm-v7`. Do you agree?
- I moved docker plugin artifacts like systemd service and plugin dockerfile to `./contrib/docker-plugin/...`. Do you agree?

**Comment by ivandeex** [comment]
@tstachl
I have opened another pull request #5668 from the branch containing your commits squashed (and having you as author) and my commit above. If you can _fix permissions_, I can close it. But if you _cannot_, let's switch our discussion there.

**Comment by maggie0002** [comment]
I have a Raspberry Pi 4 I can test it on, but not until the weekend. 

Is it just installing, mounting and checking the mount is accessible? If you have any tests you tend to run let me know and will try and stick to the same process.

**Comment by maggie0002** [comment]
> @ncw
> 
> This patch will introduce a breaking change from user's POV. Plugin will not use the `latest` tag anymore. Instead, users will have to install it by architecture-specific tags like `rclone/docker-volume-rclone:amd64` or `rclone/docker-volume-rclone:arm-v7-1.56.2`.
> 
> This **must** be noted in the changelog and in a "breaking announcements" section on the forum (do you recall its name?)
> 
> The core reason for this is that plugins a.t.m. cannot span architectures like normal multi-arch docker images.

Could the :latest tag stay there but for the current arch? Eventually I imagine it will become multi arch compatible along with the other docker infrastructure. Not sure if this will be confusing for users, but would stop it being a breaking change.

**Comment by ivandeex** [comment]
> Is it just installing, mounting and checking the mount is accessible?

Yes, that's enough

> Could the :latest tag stay there but for the current arch?
> Not sure if this will be confusing for users

I'm afraid that having `latest` === `amd64` will sooner or later bring confusion. If we merge this PR by 1.57 then 1.56.0-1.56.2 will be the only releases with `latest`. If I had time to review this earlier we could constrain the quirk even to ~1 month.
I'm waiting for @ncw decision here. He might even accept backporting this to 1.56.3 :)

**Comment by tstahl** [comment]
Thank you!  Whatever you are working on looks really interesting though :)

Cheers!
-ts

**Comment by web2brain** [comment]
> @tstachl
> @maggie0002 We are working on releasing docker plugin for non-Intel architectures. Do you have Raspberry PI boxes with docker, namely `amd-v7`? Could you give a try to beta [ivandeex/docker-volume-plugin:arm-v7](https://hub.docker.com/repository/docker/ivandeex/docker-volume-rclone/tags?page=1&ordering=last_updated)? Could you check that it installs and can do a mount:
> 
> ```
> docker plugin install ivandeex/docker-volume-rclone:arm-v7 --grant-all-permissions --alias rclone
> docker volume create testvol -d rclone -o <to your taste>
> docker run --rm -it -v testvol:/mnt -w /mnt ubuntu
> ```

I also own a Raspberry Pi 4 with docker and as I am interested in this pull request and follow it since quite some time I gave it a try - and it works as it is supposed to:

```
 user@pi:~/docker $ docker plugin install ivandeex/docker-volume-rclone:arm-v7 --grant-all-permissions --alias rclone
 arm-v7: Pulling from ivandeex/docker-volume-rclone
 Digest: sha256:4b79b240535023c1184b2f862119e5911c0040730a4cdcafdc77a4038b8e8c01
 184cad194255: Complete 
 Installed plugin ivandeex/docker-volume-rclone:arm-v7
 user@pi:~/docker $ docker volume create testvol -d rclone -o remote=dropbox: -o allow_other=true -o vfs_cache_mode=full -o poll_interval=0
 testvol
 user@pi:~/docker $ docker run --rm -it -v testvol:/mnt -w /mnt ubuntu
 Unable to find image 'ubuntu:latest' locally
 latest: Pulling from library/ubuntu
 29a0bfee4452: Pull complete
 Digest: sha256:a0d9e826ab87bd665cfc640598a871b748b4b70a01a4f3d174d4fb02adad07a9
 Status: Downloaded newer image for ubuntu:latest
 root@efc5952b057c:/mnt# ls
 Apps  Dir1  Dir2  Public  test
 root@efc5952b057c:/mnt#
 root@efc5952b057c:/mnt# echo "test" >> rclone_test.txt
 root@efc5952b057c:/mnt#
```
The created file `rclone_test.txt` shows up in Dropbox and also contains the string "test".

I hope I could help a little bit. Looking forward for this to be merged.

**Comment by ivandeex** [comment]
Closing in favor of #5668 
Will merge from #5668

**Comment by ivandeex** [comment]
@tstachl @maggie0002 @web2brain 
Thank you all for your contributions :pray:

**Comment by web2brain** [comment]
Hi @ivandeex,
I don't want to spoil the party, but it seems like the latest plug-in does not work on arm-v7 anymore.

```
user@pi:~ $ docker plugin install ivandeex/docker-volume-rclone:arm-v7 --grant-all-permissions --alias rclone
arm-v7: Pulling from ivandeex/docker-volume-rclone
Digest: sha256:171923700d87e43b9524b6a9dd0eeb0dc208fa946407bc1a6523a35350fa1c01
ecf19c14d9f8: Complete
Error response from daemon: dial unix /run/docker/plugins/a023a47cf67d94e81b102c167be2c97911daccba2b644d58b9daac382158fbe7/rclone.sock: connect: no such file or directory
```

The plug-in can't be enabled:
```
user@pi:~ $ docker plugin ls
ID             NAME            DESCRIPTION                       ENABLED
a023a47cf67d   rclone:latest   Rclone volume plugin for Docker   false
```

Enabling it manually gives the same error as the install.

Before it worked as intended. Sorry that I didn't check again sooner!

**Comment by ivandeex** [comment]
@web2brain 
Probably I spoiled the dev repo on docker hub while testing other branches.
I have just rerun the build job.
The results: https://hub.docker.com/r/ivandeex/docker-volume-rclone/tags
The build log: https://github.com/ivandeex/rclone/runs/4068209314?check_suite_focus=true
Hopefully it works for you now.

**Comment by web2brain** [comment]
@ivandeex
thanks for the rerun. Unfortunately, it did not change things and I can't get the plugin to install. I tried both ```arm-v7``` and ```arm-v7-1.57.0-beta01```.

The good news is, that I did not get the ```exec format error``` in syslog that happens when the binary format is not compatible, but nevertheless the plugin can't be enabled. I took a glance at the build log and there it is obvious that the arm version is pushed with the two tags ```arm-v7``` and ```arm-v7-1.57.0-beta01``` so the problem might be something else.

Syslog does not mention much more than that, unfortunately:
```
Nov  1 15:59:01 pi systemd[1]: Started libcontainer container 5c8b36888a0a65e928c015bad30d37e81e25911faa2471b1af31a96086785953.
Nov  1 15:59:13 pi systemd[1]: runc-5c8b36888a0a65e928c015bad30d37e81e25911faa2471b1af31a96086785953.scope: Succeeded.
Nov  1 15:59:13 pi containerd[485]: time="2021-11-01T15:59:13.975296252+01:00" level=info msg="shim disconnected" id=5c8b36888a0a65e928c015bad30d37e81e25911faa2471b1af31a96086785953
Nov  1 15:59:13 pi dockerd[29065]: time="2021-11-01T15:59:13.975364955+01:00" level=info msg="ignoring event" container=5c8b36888a0a65e928c015bad30d37e81e25911faa2471b1af31a96086785953 module=libcontainerd namespace=plugins.moby topic=/tasks/delete type="*events.TaskDelete"
Nov  1 15:59:13 pi containerd[485]: time="2021-11-01T15:59:13.980016124+01:00" level=error msg="copy shim log" error="read /proc/self/fd/67: file already closed"
Nov  1 15:59:14 pi dockerd[29065]: time="2021-11-01T15:59:14.174852515+01:00" level=error msg="Handler for POST /v1.41/plugins/rclone:latest/enable returned error: dial unix /run/docker/plugins/5c8b36888a0a65e928c015bad30d37e81e25911faa2471b1af31a96086785953/rclone.sock: connect: no such file or directory"
```

Do you have an idea what else I should try?

**Comment by ivandeex** [comment]
@web2brain 
Strange.
When I merged this, I did a manual check with amd64 and arm64.
I will now recheck again.
By the way, while we are investigating the glitch, Nick has just released 1.57.0
Do you see the problem with https://hub.docker.com/r/rclone/docker-volume-rclone/tags ?
i.e. `docker plugin install rclone/docker-volume-rclone:arm-v7 ....`
If yes, can you run `docker run -it rclone/rclone version` on your arm/v7 box?
I'm asking because plugin build script in fact repacks rclone executable from `rclone/rclone`. I wonder if the glitch is with repacking...

**Comment by web2brain** [comment]
@ivandeex
The problem persist with the latest release (```docker plugin install rclone/docker-volume-rclone:arm-v7 --grant-all-permissions --alias rclone```).

My rclone image was still version 1.56.2:
```
user@pi:~ $ docker run -it rclone/rclone version
rclone v1.56.2
- os/version: alpine 3.14.2
- os/kernel: 5.10.63-v7l+ (armv7l)
- os/type: linux
- os/arch: arm
- go/version: go1.17.1
- go/linking: static
- go/tags: none
```
After I pulled the latest version, the problem is still present:
```
user@pi:~ $ docker run -it rclone/rclone version
rclone v1.57.0
- os/version: alpine 3.14.2
- os/kernel: 5.10.63-v7l+ (armv7l)
- os/type: linux
- os/arch: arm
- go/version: go1.17.2
- go/linking: static
- go/tags: none
```

**Comment by ivandeex** [comment]
@web2brain

**tl;dr** I cannot reproduce your problem. All plugin architectures work for me!

I recommend you (1) check that prerequisite config/cache directories exist and (2) remove previously installed plugins then start cleanly
```
mkdir -p /var/lib/docker-plugins/rclone/config
mkdir -p /var/lib/docker-plugins/rclone/cache

docker volume ls
docker plugin ls

docker plugin rm rclone
etc etc
```

Here follow my tests...

### 1. Verify prerequisites

First off, I checked that prerequisite plugin directories do exist on my machine
```
root@hardy:~# ls -l /var/lib/docker-plugins/rclone/
drwxrwxr-x 3 root rclone 4096 Oct 11 19:46 cache
drwxrwxr-x 2 root rclone 4096 Nov  2 20:28 config
```

Then checked that test filesystem works fine
```
root@hardy:~# rclone --config=/var/lib/docker-plugins/rclone/config/rclone.conf config show fstest:
--------------------
[fstest]
type = http
url = https://redacted.com/somedir
--------------------
root@hardy:~# rclone --config=/var/lib/docker-plugins/rclone/config/rclone.conf ls fstest:
        8 hello.txt
```

### 2. Test on vanilla AMD64 box

Check that amd64 rclone image (**not a plugin yet**) works fine but arm64 fails:
```
root@hardy:~# docker run --platform linux/arm64 rclone/rclone:latest
latest: Pulling from rclone/rclone.......
Digest: sha256:1e6eeabddc013302793aad1986af0b8929bb42bb7c616b10a9b66b2ce04e165f
Status: Downloaded newer image for rclone/rclone:latest
standard_init_linux.go:228: exec user process caused: exec format error

root@hardy:~# docker run rclone/rclone:latest version
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
standard_init_linux.go:228: exec user process caused: exec format error

root@hardy:~# docker run --platform linux/amd64 rclone/rclone:latest version
latest: Pulling from rclone/rclone
Digest: sha256:1e6eeabddc013302793aad1986af0b8929bb42bb7c616b10a9b66b2ce04e165f
Status: Downloaded newer image for rclone/rclone:latest
rclone v1.57.0
- os/version: alpine 3.14.2 (64 bit)
- os/kernel: 4.15.0-161-generic (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.17.2
- go/linking: static
- go/tags: none
```

It was a simple rclone image.
Now checking that rclone volume plugin for AMD64 works on AMD64
```
root@hardy:~# docker plugin install rclone/docker-volume-rclone:amd64 --alias rclone --grant-all-permissions
amd64: Pulling from rclone/docker-volume-rclone
Digest: sha256:dd64dace6106c548bb75419fab36b6c04ca2394e542e10490e08ce62581f26c4
65cdd3e5660b: Complete
Installed plugin rclone/docker-volume-rclone:amd64

root@hardy:~# sudo runc --root /run/docker/runtime-runc/plugins.moby exec $(docker plugin list --no-trunc | awk '/rclone/{print$1}') rclone version
rclone v1.57.0
- os/version: alpine 3.14.2 (64 bit)
- os/kernel: 4.15.0-161-generic (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.17.2
- go/linking: static
- go/tags: none

root@hardy:~# docker volume ls
DRIVER    VOLUME NAME
root@hardy:~# docker volume create vol1 -d rclone -o fs=fstest:
vol1

root@hardy:~# docker run --rm -it -v vol1:/mnt -w /mnt ubuntu ls
Unable to find image 'ubuntu:latest' locally
latest: Pulling from library/ubuntu
7b1a6ab2e44d: Pull complete
Digest: sha256:626ffe58f6e7566e00254b638eb7e0f3b11d4da9675088f4781a50ae288f3322
Status: Downloaded newer image for ubuntu:latest
hello.txt

root@hardy:~# docker volume rm vol1
vol1
root@hardy:~# docker plugin rm --force rclone
rclone
```

Now checking that rclone volume plugin built for ARM64 fails as expected
```
root@hardy:~# docker plugin install rclone/docker-volume-rclone:arm64 --alias rclone --grant-all-permissions                                 arm64: Pulling from rclone/docker-volume-rclone
Digest: sha256:4796d2cf7c8eca95d945b64a16fcb382c3978972edfe407ef3e8fe9dcfa5281f
c06e19ea97b8: Complete
Error response from daemon: dial unix /run/docker/plugins/5cc98ef6e2a8170d83e623c95578fe10e3a1ace6ab132e34734ebeaeba4be807/rclone.sock: connect: no such file or directory

root@hardy:~# sudo runc --root /run/docker/runtime-runc/plugins.moby exec $(docker plugin list --no-trunc | awk '/rclone/{print$1}') rclone version
ERRO[0000] exec failed: container "5cc98ef6e2a8170d83e623c95578fe10e3a1ace6ab132e34734ebeaeba4be807" does not exist
```

It's all fine. But you claim you have problems with RPI and `arm/v7`...

### 3. Failed attempts to test on an ARM box

I never possessed a Raspberry Pi with arm/v7 or whatever.
I used to test arm64 stuff on a promotional OCI ampere box.
Starting the reproduction test today, I unexpectedly found that Oracle silently terminated my free-forever (_larry: haha_) Ampere VPS so I can't easily check it. Attempts to recreate the VPS failed (_chorus: mwah haha x2_).

So I just googled for a quick Qemu/Ubuntu guide, found one (https://translatedcode.wordpress.com/2017/07/24/installing-debian-on-qemus-64-bit-arm-virt-board) and two (https://lanyaplab.com/index.php/2020/06/08/how-to-install-ubuntu-server-for-arm-in-a-qemu-aarch64-virtual-machine). I tried the second guide. After an hour it was still booooting the arm64 ubuntu image so I had to find a faster way...

### 4. Test on AMD64 box with QEMU ARM emulator

The docker multi-platform build guide (https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images) recommended to run a static QEMU docker container https://github.com/tonistiigi/binfmt#installing-emulators which installs linux kernel hooks recognizing ARM binary formats and running them in a QEMU sandbox.
After that docker will support the `--platform linux/ARCH` option and run foreign architecture images.

Activate QEMU ARM emulator for Docker:
```
root@hardy:~# docker run --privileged --rm tonistiigi/binfmt --install all
Unable to find image 'tonistiigi/binfmt:latest' locally
latest: Pulling from tonistiigi/binfmt
981e1f26977e: Pull complete
6269e8ad230e: Pull complete
Digest: sha256:11128304bc582dc7dbaa35947ff3e52e2610d23cecb410ddfa381a6ce74fa763
Status: Downloaded newer image for tonistiigi/binfmt:latest
installing: arm OK
installing: arm64 OK
installing: riscv64 OK
installing: mips64le OK
installing: mips64 OK
installing: s390x OK
installing: ppc64le OK
{
  "supported": [
    "linux/amd64",
    "linux/arm64",
    "linux/riscv64",
    "linux/ppc64le",
    "linux/s390x",
    "linux/386",
    "linux/arm/v7",
    "linux/arm/v6"
  ],
  "emulators": [
    "qemu-aarch64",
    "qemu-arm",
    "qemu-mips64",
    "qemu-mips64el",
    "qemu-ppc64le",
    "qemu-riscv64",
    "qemu-s390x"
  ]
}
```

Run a simple ARM64 rclone image
```
root@hardy:~# docker run --platform linux/arm64 rclone/rclone:latest version
Unable to find image 'rclone/rclone:latest' locally
latest: Pulling from rclone/rclone
Digest: sha256:1e6eeabddc013302793aad1986af0b8929bb42bb7c616b10a9b66b2ce04e165f
Status: Downloaded newer image for rclone/rclone:latest
rclone v1.57.0
- os/version: alpine 3.14.2 (64 bit)
- os/kernel: 4.15.0-161-generic (aarch64)
- os/type: linux
- os/arch: arm64
- go/version: go1.17.2
- go/linking: static
- go/tags: none
```

Test ARM64 docker plugin
```
root@hardy:~# docker plugin rm rclone --force
rclone

root@hardy:~# docker plugin install rclone/docker-volume-rclone:arm64 --alias rclone --grant-all-permissions
arm64: Pulling from rclone/docker-volume-rclone
Digest: sha256:4796d2cf7c8eca95d945b64a16fcb382c3978972edfe407ef3e8fe9dcfa5281f
c06e19ea97b8: Complete
Installed plugin rclone/docker-volume-rclone:arm64

root@hardy:~# sudo runc --root /run/docker/runtime-runc/plugins.moby exec $(docker plugin list --no-trunc | awk '/rclone/{print$1}') rclone version
rclone v1.57.0
- os/version: alpine 3.14.2 (64 bit)
- os/kernel: 4.15.0-161-generic (aarch64)
- os/type: linux
- os/arch: arm64
- go/version: go1.17.2
- go/linking: static
- go/tags: none

root@hardy:~# docker volume create vol1 -d rclone -o fs=fstest:
vol1

root@hardy:~# docker run --rm -it -v vol1:/mnt -w /mnt ubuntu ls
hello.txt

root@hardy:~# docker volume rm vol1
vol1
```

### 5. Test the ARMv7 plugin with emulator

Run simple arm/v7 rclone image
```
root@hardy:~# docker run --platform linux/arm/v7 rclone/rclone:latest version                                                                Unable to find image 'rclone/rclone:latest' locally
latest: Pulling from rclone/rclone
a14774a5a62e: Pull complete
813a348134c5: Pull complete
ca9189655350: Pull complete
4d9bf6f89fa8: Pull complete
8c3c31937140: Pull complete
Digest: sha256:1e6eeabddc013302793aad1986af0b8929bb42bb7c616b10a9b66b2ce04e165f
Status: Downloaded newer image for rclone/rclone:latest
rclone v1.57.0
- os/version: alpine 3.14.2
- os/kernel: 4.15.0-161-generic (armv7l)
- os/type: linux
- os/arch: arm
- go/version: go1.17.2
- go/linking: static
- go/tags: none
```

Test the ARMv7 docker plugin
```
root@hardy:~# docker plugin ls
ID             NAME            DESCRIPTION                       ENABLED
6b39f60ef3bc   rclone:latest   Rclone volume plugin for Docker   true
root@hardy:~# docker plugin rm --force rclone:latest
rclone:latest

root@hardy:~# docker plugin install rclone/docker-volume-rclone:arm-v7 --alias rclone --grant-all-permissions
arm-v7: Pulling from rclone/docker-volume-rclone
Digest: sha256:281cd255a4f48e3d9d3aad44b9528dc58cf80455dfb872ed07fd14ba7adf860f
56bf270e97e7: Complete
Installed plugin rclone/docker-volume-rclone:arm-v7

root@hardy:~# sudo runc --root /run/docker/runtime-runc/plugins.moby exec $(docker plugin list --no-trunc | awk '/rclone/{print$1}') rclone version
rclone v1.57.0
- os/version: alpine 3.14.2
- os/kernel: 4.15.0-161-generic (armv7l)
- os/type: linux
- os/arch: arm
- go/version: go1.17.2
- go/linking: static
- go/tags: none

root@hardy:~# docker volume create vol1 -d rclone -o fs=fstest:
vol1

root@hardy:~# docker run --rm -it -v vol1:/mnt -w /mnt ubuntu ls
hello.txt

root@hardy:~# docker volume rm vol1
vol1
root@hardy:~# docker plugin rm --force rclone:latest
rclone:latest
```

Success!
cc @ncw

**Comment by web2brain** [comment]
@ivandeex tl;dr Sorry, it was only a local problem (pebkac :) ).

The needed files and directories were all there and the filesystems in ```/var/lib/docker-plugins/rclone/config``` were recognised as thez should.
What solved it in the end was to empty ```/var/lib/docker-plugins/rclone/cache``` before installing the plugin.

I'm sorry for your additional effort but happy, that at least I did not postpone the release of 1.57!

**Comment by ivandeex** [comment]
Solved
